pip/compat.py

```diff
@@ -36,22 +41,65 @@
         cache_from_source = None
 
 
-if sys.version_info >= (3,):
-    def console_to_str(s):
-        try:
-            return s.decode(sys.__stdout__.encoding)
-        except UnicodeDecodeError:
-            return s.decode('utf_8')
+if sys.version_info > (3, 4):
+    backslashreplace_decode = "backslashreplace"
+else:
+    def backslashreplace_decode_fn(err):
```

Comment:
Probably worth adding a comment here to say that while backslashreplace exists on these earlier versions, it's only usable for encoding, so this implements a version that specifically handles decoding.